core/server/common/src/main/java/alluxio/master/journal/JournalUtils.java

```diff
@@ -130,9 +129,8 @@ public static void restoreFromCheckpoint(InputStream input,
       List<? extends Checkpointed> components) throws IOException {
     CheckpointInputStream cis = new CheckpointInputStream(input);
     InputChunked chunked = new PatchedInputChunked(input);
-    if (cis.getType() != CheckpointType.COMPOUND) {
-      throw new IllegalStateException("Unexpected checkpoint type: " + cis.getType());
-    }
+    Preconditions.checkState(cis.getType() != CheckpointType.COMPOUND,
```

Comment:
`!=` -> `==`

Reply:
just wanted to show off our amazing test coverage :D